tests/digest_validation.rs

```diff
@@ -0,0 +1,279 @@
+use std::net::SocketAddr;
+
```

Comment:
nitpicking, the empty line can be removed. Also, some comment at the top of the file explaining what is going on would be nice

Reply:
Actually the line should be there. The common pattern for separating imports in order is 

- core
- std
- external crates
- internal crate imports

Otherwise they're just all smooshed together.

I honestly wish this was a cargo fmt thing but it isn't. I do know that when rust-analyzer adds imports it separates them as well last time I dug into it

Reply:
Thanks for the explanation, I didn't realize that.